code/controllers/subsystem/blackbox.dm

```diff
@@ -0,0 +1,274 @@
+	json: {"data":["sample text","other text"]}
+"amount"
+	used to record simple counts of data i.e. the number of ahelps recieved
+	further calls to the same key will add or subtract (if increment argument is a negative) from the saved amount
+	calls:	SSblackbox.record_feedback("amount", "example", 8)
+			SSblackbox.record_feedback("amount", "example", 2)
+	json: {"data":10}
+"tally"
+	used to track the number of occurances of multiple related values i.e. how many times each type of gun is fired
+	further calls to the same key will:
+	 	add or subtract from the saved value of the data key if it already exists
+		append the key and it's value if it doesn't exist
+	calls:	SSblackbox.record_feedback("tally", "example", 1, "sample data")
+			SSblackbox.record_feedback("tally", "example", 4, "sample data")
+			SSblackbox.record_feedback("tally", "example", 2, "other data")
+	json: {"data":{"sample data":5,"other data":2}}
+"nested tally"
+	used to track the number of occurances of structured semi-relational values i.e. the results of arcade machines
+	similar to running total, but related values are nested in a multi-dimensional array built
+	the final element in the data list is used as the tracking key, all prior elements are used for nesting
+	further calls to the same key will:
+	 	add or subtract from the saved value of the data key if it already exists in the same multi-dimensional position
+		append the key and it's value if it doesn't exist
+	calls: 	SSblackbox.record_feedback("nested tally", "example", 1, list("fruit", "orange", "apricot"))
+			SSblackbox.record_feedback("nested tally", "example", 2, list("fruit", "orange", "orange"))
+			SSblackbox.record_feedback("nested tally", "example", 3, list("fruit", "orange", "apricot"))
+			SSblackbox.record_feedback("nested tally", "example", 10, list("fruit", "red", "apple"))
+			SSblackbox.record_feedback("nested tally", "example", 1, list("vegetable", "orange", "carrot"))
+	json: {"data":{"fruit":{"orange":{"apricot":4,"orange":2},"red":{"apple":10}},"vegetable":{"orange":{"carrot":1}}}}
+	tracking values associated with a number can't merge with a nesting value, trying to do so will append the list
+	call:	SSblackbox.record_feedback("nested tally", "example", 3, list("fruit", "orange"))
+	json: {"data":{"fruit":{"orange":{"apricot":4,"orange":2},"red":{"apple":10},"orange":3},"vegetable":{"orange":{"carrot":1}}}}
+"associative"
+	used to record text that's associated with a value i.e. coordinates
+	further calls to the same key will append a new list to existing data
+	calls:	SSblackbox.record_feedback("associative", "example", 1, list("text" = "example", "path" = /obj/item, "number" = 4))
+			SSblackbox.record_feedback("associative", "example", 1, list("number" = 7, "text" = "example", "other text" = "sample"))
+	json: {"data":{"1":{"text":"example","path":"/obj/item","number":"4"},"2":{"number":"7","text":"example","other text":"sample"}}}
```

Comment:
This is 
```json
 {"data":{"1":{"text":"example","path":"/obj/item","number":"4"},"2":{"number":"7","text":"example","other text":"sample"}}}
 ```
 and not 
 ```json
  {"data": [{"text":"example","path":"/obj/item","number":"4"}, {"number":"7","text":"example","other text":"sample"}]}
  ```
  only for technical reasons, and not because it's a better way, right?

Also, likely related, what does `1` in the call stand for?

Reply:
If you read the comment, its for associative data. `1` is a key here, not an index.

Reply:
Then why does `SSblackbox.record_feedback("associative", "example", 1, list("number" = 7, "text" = "example", "other text" = "sample"))` add `{"number":"7","text":"example","other text":"sample"}` under the key of `2`, not `1`?

Reply:
Because its assosiactive with `"1"` and `"2"`, not index 1 and 2

Reply:
But the `"2"` (nor `2`) is not specified anywhere in the in the
```dm
SSblackbox.record_feedback("associative", "example", 1, list("number" = 7, "text" = "example", "other text" = "sample"))
```
call!

Reply:
Because it's associative to the first and second calls? There's 2 procs there.